pkg/server/image_pull_test.go

```diff
@@ -93,3 +95,49 @@ func TestResources(t *testing.T) {
 		assert.True(t, ok)
 	}
 }
+
+func TestParseAuth(t *testing.T) {
+	testUser := "username"
+	testPasswd := "password"
+	testAuthLen := base64.StdEncoding.EncodedLen(len(testUser + ":" + testPasswd))
+	testAuth := make([]byte, testAuthLen)
+	base64.StdEncoding.Encode(testAuth, []byte(testUser+":"+testPasswd))
```

Comment:
maybe add a bad auths here to get closer to 100% test coverage on the api

For example:
invalidAuthLen := base64.StdEncoding.EncodedLen(len(testUser + "%" + testPasswd))
invalidDecodeAuth := make([]byte, invalidAuthLen)
base64.StdEncoding.Encode(invalidDecodeAuth, []byte(testUser+"%"+testPasswd)) // % is an invalid base64 char...and an invalid separator